packages/webview_flutter/lib/platform_interface.dart

```diff
@@ -0,0 +1,44 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/gestures.dart';
+import 'package:flutter/widgets.dart';
+
+/// Interface for talking to the webview controller on the platform side.
+/// 
+/// An instance implementing this interface is passed to the `onWebViewCreated` callback that is
+/// passed to [WebViewPlatformInterface#onWebViewCreated].
+abstract class WebViewPlatformControllerInterface {
+  Future<void> loadUrl(
+      String url,
+      Map<String, String> headers,
+      );
+
+  // As the PR currently focus about the wiring I've only moved loadUrl to the new way, so
+  // the discussion is more focused.
+  // In this temporary state WebViewController still uses a method channel directly for all other
+  // method calls so we need to expose the webview ID.
+  // TODO(amirh): remove this before submitting this PR(after getting an LGTM for the overall approach).
+  int get id;
+}
+
+typedef WebViewCreatedCallback = void Function(WebViewPlatformControllerInterface);
```

Comment:
nit: we like to name the arguments in typedefs. In some places, without a name, the type becomes the name (e.g. `void bar(Foo)` means `void bar(dynamic Foo)`) so it's better to be consistent throughout.

Reply:
done